packages/core/src/codewhispererChat/controllers/chat/controller.ts

```
@@ -930,52 +957,58 @@ export class ChatController {
             )
             session.relativePathToWorkspaceRoot.set(relativePath, contextCommand.workspaceFolder)
         }
-        const prompts = await LspClient.instance.getContextCommandPrompt(contextCommands)
-        if (prompts.length === 0) {
-            return []
+        let prompts: AdditionalContextPrompt[] = []
+        try {
+            prompts = await LspClient.instance.getContextCommandPrompt(contextCommands)
+        } catch (e) {
+            getLogger().verbose(`Could not get context command prompts: ${e}`)
```

swallowing errors are dangerous. So unless you have a good reason they should propagate upwards and be handled properly if possible

---

Its not an error that I want to throw or surface. this function fails if the user tries to use `@workspace` but indexing is not yet complete. we have a follow up item to better address this case but for now we don't want to throw an error. I can add a todo here

---

It throws when LSP is not ready (not spinning up yet) or the LSP somehow went wrong. In this case, we return empty list and swallow the error so that the user can still see the context menu popping up.

---

If we throw here, the UX will be broken. 

In the near short term, this API will be moved to FLARE.

---

I see, sounds good

If you see value, you can emit a metric on failure to have some data to observe in Kibana. Just a recomendation.